substrate/frame/identity/src/benchmarking.rs

```diff
@@ -118,110 +122,124 @@ fn create_identity_info<T: Config>(num_fields: u32) -> IdentityInfo<T::MaxAdditi
-			let caller: T::AccountId = whitelisted_caller();
-			let caller_lookup = T::Lookup::unlookup(caller.clone());
-			let caller_origin: <T as frame_system::Config>::RuntimeOrigin = RawOrigin::Signed(caller.clone()).into();
-			let _ = T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
-
-			// Add an initial identity
-			let initial_info = create_identity_info::<T>(1);
-			Identity::<T>::set_identity(caller_origin.clone(), Box::new(initial_info.clone()))?;
-
-			// User requests judgement from all the registrars, and they approve
-			for i in 0..r {
-				let registrar: T::AccountId = account("registrar", i, SEED);
-				let registrar_lookup = T::Lookup::unlookup(registrar.clone());
-				let balance_to_use =  T::Currency::minimum_balance() * 10u32.into();
-				let _ = T::Currency::make_free_balance_be(&registrar, balance_to_use);
-
-				Identity::<T>::request_judgement(caller_origin.clone(), i, 10u32.into())?;
-				Identity::<T>::provide_judgement(
-					RawOrigin::Signed(registrar).into(),
-					i,
-					caller_lookup.clone(),
-					Judgement::Reasonable,
-					T::Hashing::hash_of(&initial_info),
-				)?;
-			}
-			caller
-		};
-	}: _(RawOrigin::Signed(caller.clone()), Box::new(create_identity_info::<T>(x)))
-	verify {
+	#[benchmark]
+	fn set_identity(
+		r: Linear<1, { T::MaxRegistrars::get() }>,
+		x: Linear<0, { T::MaxAdditionalFields::get() }>,
+	) -> Result<(), BenchmarkError> {
+		add_registrars::<T>(r)?;
+
+		let caller: T::AccountId = whitelisted_caller();
+		let caller_lookup = T::Lookup::unlookup(caller.clone());
+		let caller_origin: <T as frame_system::Config>::RuntimeOrigin =
+			RawOrigin::Signed(caller.clone()).into();
+		let _ = T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+		// Add an initial identity
+		let initial_info = create_identity_info::<T>(1);
+		Identity::<T>::set_identity(caller_origin.clone(), Box::new(initial_info.clone()))?;
+
+		// User requests judgement from all the registrars, and they approve
+		for i in 0..r {
+			let registrar: T::AccountId = account("registrar", i, SEED);
+			let registrar_lookup = T::Lookup::unlookup(registrar.clone());
+			let balance_to_use = T::Currency::minimum_balance() * 10u32.into();
+			let _ = T::Currency::make_free_balance_be(&registrar, balance_to_use);
+
+			Identity::<T>::request_judgement(caller_origin.clone(), i, 10u32.into())?;
+			Identity::<T>::provide_judgement(
+				RawOrigin::Signed(registrar).into(),
+				i,
+				caller_lookup.clone(),
+				Judgement::Reasonable,
+				T::Hashing::hash_of(&initial_info),
+			)?;
+		}
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller.clone()), Box::new(create_identity_info::<T>(x)));
 		assert_last_event::<T>(Event::<T>::IdentitySet { who: caller }.into());
+		Ok(())
 	}
 
-	// We need to split `set_subs` into two benchmarks to accurately isolate the potential
-	// writes caused by new or old sub accounts. The actual weight should simply be
-	// the sum of these two weights.
```

Comment:
This comment seems helpful. Why remove it?

Reply:
Ah, accidental. Will bring it back.